doc/docker.md

```diff
@@ -1,111 +1,163 @@
+---------------------------
 
+We run a continuous integregation testing pipe using [Travis CI](https://travis-ci.com/). This runs three containers which are managed in the docker-compose.test.yml file.
 
-**Web Server Container**
+ - **webserver** nginx server that contains the frontend static files, built with webpack
+ - **expungeservice** wsgi server (hosting a Flask app)
+ - **db** postgres database
 
-This container will run the Nginx web server, which serves the static frontend pages, and forwards api calls to the backend service.
+Prod Environment Containers
+---------------------------
 
+The app is deployed to Heroku in just two containers, each served from a standalone Heroku app:
 
-**Database Container**
+ - **webserver** The frontend is an nginx container that serves the static files, and proxies api requests to the backend. It runs at https://recordexpungpdx.herokuapp.com/
+ - **expungeservice** The backend is a wsgi server that runs at https://recordexpungpdxapi.herokuapp.com/
 
-This container will run the PostgreSQL database.
+The database runs in a heroku-postgres plugin that is attached to the backend.
 
+Dockerfiles
+-----------
 
-**Service Container**
+Some variations exist between the Dockerfiles used for each enviroment. TODO: elaborate on this a little bit.
 
-This container will run the Python/Flask backend.
 
+Running the Dev containers
+--------------------------
 
-Running the containers
-----------------------
+Launching and managing the local workspace uses a few written in the project's Makefile:
 
-Development of component source code should often be possible without touching Dockerfiles, so here are a few useful commands that might be sufficient for running and updating docker images in the dev environment.
+Launch the docker enviroment with, and then launch the containers with.
+
+```
+make dev_up
+```
+
+Thist first builds the dev docker images if they don't already exist. It runs the docker stack in detached mode, so you can follow the container of logs with:
+
+```
+make dev_logs
+```
+
+While the dev stack is running, each service is accessible in the dev environment at `localhost:<port>`, at the exposed ports defined in the docker-compose.dev.yml file.
+
+Stop the running containers with
+
+```
+make dev_down
+```
+
+To force a rebuild of the docker images (if you change the dependencies), use
+
+```
+make dev_logs
+
+```
+
+Some Docker CLI commands
+------------------------
+
+To see which containers are running (and with the optional flag to see stopped containers also):
+
+
+        docker ps [-a]
+
+
+Running the Test containers
+---------------------------
+
+Because of the differences between the `dev` and `test` stacks, a different set of Dockerfiles and commands are used to build and run the test containers. You can do this for local testing as well as with the automated scripts on Travis. The deployment uses `docker stack` commands rather than `docker-compose`. This might be worth changing?
```

Comment:
We could consider using `docker-compose` when running this with Travis, for the sake of simplifying things.

Reply:
Cool, let's talk about it